File: 0118-pascals-triangle/0118-pascals-triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int n) {
        vector<vector<int>> arr;
  for(int i=0;i<n;i++){
     arr.push_back(vector<int>(i + 1));
        for(int j=0;j<=i;j++){
            if(j==0 || j==i){
                arr[i][j]=1;

            }
            else{
                arr[i][j]=arr[i-1][j]+arr[i-1][j-1];
            }

        }

        }
        return arr;
    }
};

File: leetcode/0118-pascals-triangle/0118-pascals-triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:

        arr=[]
        for rows in range(numRows):
            in_arr=[]
            for cols in range(rows+1):

                
                if(cols==0 or  cols==rows):
                    in_arr.append(1)
                else:
                    val=arr[rows-1][cols]+arr[rows-1][cols-1]
                    in_arr.append(val)
            arr.append(in_arr)

        return arr


        